scipy/special/basic.py

```diff
@@ -1113,15 +1113,59 @@ def hyp0f1(v, z):
         res = np.empty_like(z, dtype=np.float64)
     poles = (v < 0) & (v == v.astype(int))
     zeros = (z == 0)
-    pos = (z.real >= 0) & -zeros & -poles
-    neg = (z.real < 0) & -zeros & -poles
+    pos = (z.real >= 0) & ~zeros & ~poles
+    neg = (z.real < 0) & ~zeros & ~poles
+    asy = v > 150
+    pos_asy = pos & asy
+    pos = pos & ~asy
+
     res[poles] = nan
     res[zeros] = 1
     res[pos] = z[pos]**((1.0 - v[pos])/2)*gamma(v[pos])*iv(v[pos] - 1, 2*sqrt(z[pos]))
     res[neg] = (-z[neg])**((1.0 - v[neg])/2)*gamma(v[neg])*jv(v[neg] - 1, 2*sqrt(-z[neg]))
+
+    if np.any(pos_asy):
+        res[pos_asy] = _hyp0f1_asy(v[pos_asy], z[pos_asy])
+
     return res
 
 
+def _hyp0f1_asy(v, z):
+    r"""Asymptotic expansion for Iv[z] for real $z$ and $v\to +\infty$.
+
+    Based off DLMF 10.41
+    """
+    den = np.sqrt(abs(z))**(v - 1.0)
+    
+    v1 = v - 1
+    x = 2 * np.sqrt(abs(z)) / v1
+    
+    ## I_v(x) * Gamma(v+1), w/ DLMF 10.41 for I_{v}(vx)
+    p1 = np.sqrt(1. + x**2)
+    eta = p1 + np.log(x / (1 + p1))
+    arg_exp = v1 * eta - 0.5*np.log(p1)
+    
+    # mul by Gamma(v+1)
+    arg_exp += gammaln(v1 + 1)
```

Comment:
This is `v`. I'll push the updated version when I'm done fixing several other glitches smoked out by test_mpmath.